### ISSUE TITLE
Remove custom rule for deprecation warnigns in __main__

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#11333]
See https://www.python.org/dev/peps/pep-0565/ we might be able to remove some custom code on 3.7+ and/or have a behavior which is closer to what is defined in the pep.

**Comment by Carreau** [comment]
See #8478 and #8480 for context/historical discussions.

**Comment by yo-ga** [comment]
@Carreau I  would like to fix it. BTW, if I get it right, I just need to remove the code added in #8480

**Comment by Carreau** [comment]
> @Carreau I would like to fix it. BTW, if I get it right, I just need to remove the code added in #8480

As this is fixed in CPython only as 3.7, it way need to be moved to a : 

```
if sys.info_version < (3,7):
   ... code
```

So that once we drop 3.6 we can remove the code.

**Comment by LucianaMarques** [comment]
Hi, 

Are there any news on this issue? Was there any pull request already made?

**Comment by Carreau** [comment]
> Are there any news on this issue? Was there any pull request already made?

I don't believe any pull request was made. Feel free to take it.

**Comment by LucianaMarques** [comment]
I believe that I have a Pull Request already, however I have some doubts about the project's code and haven't ran any tests. Should I make it anyway and discuss it in the PR? Or should we solve it here?

I simply added the 

if sys.info_version < (3,7):
...code

mentioned above on the init_deprecation_warnings() functions in the main IPython class. What i understood is that this issue is fixed for python 3.7 but not for older versions, so I guessed this should suffice. 

I wanted opinions from more experienced developers in this project to help to check it and to test it. 

(I made the [PR](https://github.com/ipython/ipython/pull/11396) anyway, I saw in [Contributing.md](https://github.com/ipython/ipython/blob/6.2.x/CONTRIBUTING.md) that any PR should be done as soon as there is code to discuss)

**Comment by Carreau** [comment]
> Should I make it anyway and discuss it in the PR?
> I made the [PR](https://github.com/ipython/ipython/pull/11396) anyway

Yes, that is the right thing to do. You can run test locally with `iptest` , but making the PR will have out continuous integration suite run the tests. 

Thanks, I'll have a look at the PR.

**Comment by Carreau** [comment]
The code is now conditionally run on < 3.7 (see #11396), once Python 3.6 is dropped this can actually be removed.

**Comment by Carreau** [comment]
Thanks @LucianaMarques !
